src/UserContext/AnonymousRequestMatcher.php

```diff
@@ -13,36 +13,51 @@
 
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Matches anonymous requests using a list of identification headers.
  */
 class AnonymousRequestMatcher implements RequestMatcherInterface
 {
-    private $sessionNamePrefix;
+    /**
+     * The options configured in the constructor argument or default values.
+     *
+     * @var array
+     */
+    private $options;
 
     /**
-     * @param string $sessionNamePrefix Prefix for session cookies. Must match your PHP session configuration
+     * When creating this matcher, you can configure a number of options.
+     *
+     * - user_identifier_headers: List of request headers that authenticate a non-anonymous request.
+     * - session_name_prefix:     Prefix for session cookies. Must match your PHP session configuration.
+     *
+     * @param array $options Options to overwrite the default options
```

Comment:
there are no defaults (which is good). lets instead say "Configuration for the matcher. All options are required because this matcher is usually created by the UserContextSubscriber which provides the default values."